### PR TITLE
WFLY-4515 Fix clustering transformers and reenable in mixed-domain tests

### PR DESCRIPTION
**Pull request by rhusar**
Jira
https://issues.jboss.org/browse/WFLY-4515

### CODE REVIEW

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/StoreResourceDefinition.java [hunk 1]
```diff
@@ -137,6 +138,8 @@ public ModelNode transformOperation(ModelNode operation) {
                 }
             };
             builder.addRawOperationTransformationOverride(MapOperations.MAP_PUT_DEFINITION.getName(), new SimpleOperationTransformer(removePropertyTransformer));
+
+            builder.getAttributeBuilder().setDiscard(DiscardAttributeChecker.ALWAYS, PROPERTIES);
```
Comment (kabir) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/StoreResourceDefinition.java]: If it is possible to define a 3.0 configuration without properties, we should reject if defined

File: clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/ProtocolResourceDefinition.java [hunk 2]
```diff
@@ -184,6 +184,8 @@ public ModelNode transformOperation(ModelNode operation) {
                 }
             };
             builder.addRawOperationTransformationOverride(MapOperations.MAP_PUT_DEFINITION.getName(), new SimpleOperationTransformer(removePropertyTransformer));
+
+            builder.getAttributeBuilder().setDiscard(DiscardAttributeChecker.ALWAYS, PROPERTIES);
```
Comment (kabir) [inline clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/ProtocolResourceDefinition.java]: Reject if defined?

File: clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/StoreResourceDefinition.java (hunk 1 above)
Comment (kabir) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/StoreResourceDefinition.java]: Hmm, or perhaps the operation transformer on line 115 is mapping this onto a parameter?

Comment (rhusar) [inline clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/StoreResourceDefinition.java]: Good catch, the operation transformer does not kick in. We will need an attribute converter to be kind here.

**Comment by rhusar** [comment]
Deleting apparently closes the PR and I cannot reopen this one, so opened https://github.com/wildfly/wildfly/pull/7395 instead.

**Pull request closed without merging**